### PR TITLE
Feature/private route

### PR DESCRIPTION
**Pull request by Nicolas-Rodriguez-Ch**
### Summary 

- adds conditionally rendered buttons to the login form if the user is logged in
- adds conditionally rendered buttons if the user is any kind of admin
- adds admin routes privatization
- adds 404 error to the user page if the token is expired or the user does not exist.

### To-do

- add 404 error to admins' pages
- hydrate admin pages with the backend information
- add a connection to the back end to forms in the admin pages
- improve design and style of admin pages-

### CODE REVIEW

File: src/components/LogInComponent/LogInComponent.jsx [hunk 1]
```diff
@@ -1,48 +1,98 @@
...
 import "./LogInComponent.css";
 import languageSelector from '../../assets/languages/languageSelector';
+import { setUser } from '../../store/actions/user.action';
 
 const LogInComponent = ({ setShowLogIn }) => {
 
   const language = useSelector(state=> state.languageReducer);
-
+  const user = useSelector(state => state.userReducer.user);
+  console.log("🚀 ~ file: LogInComponent.jsx:16 ~ LogInComponent ~ user:", user)
```
Comment (ht1204) [inline src/components/LogInComponent/LogInComponent.jsx]: is this console.log necessary?

File: src/components/LogInComponent/LogInComponent.jsx [hunk 2]
```diff
@@ -1,48 +1,98 @@
...
-          </span>
+    <>
+      <ToastContainer/>
+      <div className="log-in-component">
+        <div
+          className="log-in-component-background"
+          onClick={() => setShowLogIn(false)}
+        ></div>
+        <div className="log-in-form">
+          <div className="log-in-form-title-box">
```
Comment (ht1204) [inline src/components/LogInComponent/LogInComponent.jsx:42]: Remove blankspace

File: src/components/LogInComponent/LogInComponent.jsx [hunk 3]
```diff
@@ -1,48 +1,98 @@
...
+                    {languageSelector(language, 'userPage')}
+                  </button>
+                  {isAdmin && (
+                    <button
+                      className="log-in-main-form-buttons"
+                      onClick={() =>
+                        navigate(
+                          user.user_role === "restaurantAdmin"
+                          ? "/restaurant-admin"
+                          : "/general-admin"
```
Comment (ht1204) [inline src/components/LogInComponent/LogInComponent.jsx]: These routes could be better as object prop or constants

File: src/components/LogInComponent/LogInComponent.jsx [hunk 4]
```diff
@@ -1,48 +1,98 @@
...
+                        )
+                      }
+                    >
+                    {languageSelector(
+                      language,
+                      user.user_role === "restaurantAdmin"
+                      ? "restaurantAdminPage"
+                      : "appAdminPage"
```
Comment (ht1204) [inline src/components/LogInComponent/LogInComponent.jsx:69]: same here.

File: src/pages/UserPage/UserPage.jsx [hunk 5]
```diff
@@ -84,7 +86,7 @@ const UserPage = () => {
     setIsEditable(!isEditable);
   }
 
-  // hangles the changes of the inputs
+  // handles the changes of the inputs
   const handleInputChange = (event) => {
     const { name, value } = event.target;
     const matches = name.match(/^(\w+)\[(\d+)\](\[\w+\])?$/);
```
Comment (ht1204) [inline src/pages/UserPage/UserPage.jsx]: This regex could be better in a constant imported as resource in the project.

File: src/components/LogInComponent/LogInComponent.jsx (hunk 1 above)
Comment (Nicolas-Rodriguez-Ch) [reply src/components/LogInComponent/LogInComponent.jsx]: No... resolved.

File: src/components/LogInComponent/LogInComponent.jsx (hunk 2 above)
Comment (Nicolas-Rodriguez-Ch) [reply src/components/LogInComponent/LogInComponent.jsx:42]: Resolved

File: src/components/LogInComponent/LogInComponent.jsx (hunk 3 above)
Comment (Nicolas-Rodriguez-Ch) [reply src/components/LogInComponent/LogInComponent.jsx]: Resolved, created a new folder and a document with all the routes

File: src/components/LogInComponent/LogInComponent.jsx (hunk 4 above)
Comment (Nicolas-Rodriguez-Ch) [reply src/components/LogInComponent/LogInComponent.jsx:69]: These are for the language they are not routes, should I still change them?

File: src/pages/UserPage/UserPage.jsx (hunk 5 above)
Comment (Nicolas-Rodriguez-Ch) [reply src/pages/UserPage/UserPage.jsx]: resolved

**Comment by Nicolas-Rodriguez-Ch** [comment]
### Summary

- adds constants folder in src
-  created path routes document with all the routes and started the migration of the routes to it.
- created a regex contact folder and started the migration to it.

File: src/components/LogInComponent/LogInComponent.jsx (hunk 4 above)
Comment (ht1204) [reply src/components/LogInComponent/LogInComponent.jsx:69]: let's keep it there then, np.

**Review by ht1204 — APPROVED** [review]
Great job!

**Pull request merged**